libsawtooth/src/client/rest.rs

```diff
@@ -149,6 +153,118 @@ impl SawtoothClient for RestApiSawtoothClient {
 
         Ok(get::<StatusList>(&url, error_msg)?.map(convert_status_list))?.transpose()
     }
+    /// Send one or more batches to the sawtooth REST API to be submitted to the validator
+    fn submit_batches(
+        &self,
+        filename: String,
+        wait: Option<Duration>,
+        size_limit: usize,
+    ) -> Result<(), SawtoothClientError> {
+        let url = format!("{}/batches", &self.url);
+
+        let mut batch_file = File::open(filename).map_err(|err| {
+            SawtoothClientError::new_with_source("failed to open batch file", err.into())
+        })?;
+
+        let mut contents = vec![];
+        batch_file.read_to_end(&mut contents).map_err(|err| {
+            SawtoothClientError::new_with_source("failed to read file", err.into())
+        })?;
+        let batch_list: BatchList = protobuf::parse_from_bytes(&contents).map_err(|err| {
+            SawtoothClientError::new_with_source("unable to parse file contents", err.into())
+        })?;
```

Comment:
You should be able to skip the `contents` buffer by using `protobuf::parse_from_reader(&mut batch_file)`

Reply:
Fixed